src/ui/components/CollectionsStructure/CollectionStructureDialog/CollectionStructureDialog.tsx

```diff
@@ -155,24 +156,29 @@ export const CollectionStructureDialog = React.memo<Props>(
             onClose(structureChanged);
         }, [structureChanged, onClose]);
 
+        const includePermissionsInfo = workbookSelectionMode;
+
         const getStructureItemsRecursively = React.useCallback(
             (args: GetStructureItemsArgs): CancellablePromise<GetStructureItemsResponse | null> => {
                 let curPage = args.page;
 
-                return dispatch(getStructureItems(args)).then((result) => {
-                    if (result?.items.length === 0 && result.nextPageToken !== null) {
-                        curPage = result.nextPageToken;
-
-                        return getStructureItemsRecursively({
-                            ...args,
-                            page: curPage,
-                        });
-                    } else {
-                        return result;
-                    }
-                });
+                return dispatch(getStructureItems({...args, includePermissionsInfo})).then(
+                    (result) => {
+                        if (result?.items.length === 0 && result.nextPageToken !== null) {
```

Comment:
it was already here, I just wanted to know in which case we might have an empty list and not empty the nextPageToken so that we would like to call recursion?

Reply:
In case we don't have access to all entries on the current page.